### PR TITLE
addons/pinniped/post-deploy: add --pinniped-api-group-suffix

### PR DESCRIPTION
**Pull request by ankeesler**
**What this PR does / why we need it**:

* Context: we (`area/iam`) are in the process of updating the Pinniped package and related components to support Pinniped being run in a non-default API group so that the Tanzu version of Pinniped running on the cluster will not conflict with any other version of Pinniped
* This PR updates the Pinniped post-deploy job code to take in the API group as an argument

**Which issue(s) this PR fixes**:

None at the moment...

**Describe testing done for PR**:

* Added a bunch of unit tests in the post-deploy job code
* I validated that the new post-deploy job 1) works the same way it currently does in the system (i.e., talks to the default Pinniped API group) when not passing any flags, 2) fails when we tell it to use an incorrect API group, and 3) works the same way it currently does in the system when passing a flag with the correct API group

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```
**New PR Checklist**

- [X] Ensure PR contains only public links or terms
- [X] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [X] Squash the commits in this branch before merge to preserve our git history
- [X] If this PR is just an idea or POC, use a [Draft PR](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) instead of a full PR
- [ ] Add appropriate [kind label](../docs/release/kind-labels.md) according to what type of issue is being addressed.

### CODE REVIEW

File: addons/pinniped/post-deploy/pkg/vars/vars.go [hunk 1]
```diff
@@ -5,6 +5,11 @@
 package vars
 
 var (
+	// PinnipedAPIGroupSuffix is the API group suffix used to talk to Pinniped APIs
+	// E.g., if PinnipedAPIGroupSuffix is set to "tuna.io", then FederationDomain API group would be
+	// "config.supervisor.tuna.io".
+	PinnipedAPIGroupSuffix = "pinniped.dev"
```
[the remark this answers is not in the document]
Comment (ankeesler) [reply addons/pinniped/post-deploy/pkg/vars/vars.go:11]: oh yeah sure...lemme see if i can come up with something...

[the remark this answers is not in the document]
Comment (ankeesler) [reply addons/pinniped/post-deploy/pkg/vars/vars.go]: alright i stuck that constant in `pkg/constants`. i incorporated it into 54fb1561f2e29927338a2c87d18a5119f0d68d96. wdyt?

[the remark this answers is not in the document]
Comment (benjaminapetersen) [reply addons/pinniped/post-deploy/pkg/vars/vars.go]: 👍

File: addons/pinniped/post-deploy/pkg/pinnipedclientset/pinnipedclientset.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	"k8s.io/client-go/dynamic"
+)
+
+// translateAPIGroup translates the provided pinniped baseAPIGroup to a new API group based on the
+// provided newAPIGroupSuffix.
+//
+// We assume that all baseAPIGroup's will end in "pinniped.dev", and therefore we can safely replace
+// the reference to "pinniped.dev" with the provided newAPIGroupSuffix. If the provided baseAPIGroup
+// does not end in "pinniped.dev", then this function will return an empty string.
+func translateAPIGroup(baseAPIGroup, newAPIGroupSuffix string) string {
```
Comment (sabbey37) [inline addons/pinniped/post-deploy/pkg/pinnipedclientset/pinnipedclientset.go:30]: Should we add a test for this path (when `baseAPIGroup` doesn't contain `pinniped.dev`)?

**Pull request merged**